source/classes.tex

```diff
@@ -6592,12 +6592,12 @@
 };
 
 constexpr A g() {
-  A a;
-  return a;
+  A loc;
+  return loc;
 }
 
 constexpr A a;          // well-formed, \tcode{a.p} points to \tcode{a}
-constexpr A b = g();    // well-formed, \tcode{b.p} points to \tcode{b}
+constexpr A b = g();    // error, \tcode{b.p} would point to an automatic object
```

Comment:
Where exactly is the "error" (as in: required diagnostic?)
At best, we can say that b.p points to storage whose duration has ended, right? There isn't even undefined behavior here per se, because nobody has looked at b.p, yet.

Reply:
I think your wording is better. However, the initialization of `b` is ill-formed because the expression `g()` in its initializer can't be a constant expression while appearing in constant initialization.
With NRVO forbidden, the return value of `g()` contains a pointer to a storage with automatic storage duration that has ended, which makes `g()` unable to be a constant expression.
So even though the dangling pointer hasn't be dereferenced (i.e. undefined behavior doesn't occur), it can't comprose a constant expression or be the value of a constexpr varaible.